File: 0551-Student Attendance Record I/0551-Student Attendance Record I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int countA = 0, countL = 0;
        for (char c : s) {
            if (c == 'A') {
                if (++countA > 1) {
                    return false;
                }
                countL = 0;
            }
            else if (c == 'L') {
                if (++countL > 2) {
                    return false;
                }
            }
            else { // 'P'
                countL = 0;
            }
        }
        
        return true;
    }
};

File: 0501-0600/0551-Student Attendance Record I/0551-Student Attendance Record I.py
class Solution:
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        countA = countL = 0
        for c in s:
            if c == 'A':
                countA += 1
                if countA > 1:
                    return False
                countL = 0
            elif c == 'L':
                countL += 1
                if countL > 2:
                    return False
            else:
                countL = 0
        return True
